messages.go

```diff
@@ -1,11 +1,9 @@
 package agency
 
-import "encoding/json"
-
 type Message interface {
 	Role() Role
 	Content() []byte
-	Kind() Kind
+	Kind() Kind // do we need this?
```

Comment:
We don't use this anywhere except one example just to print in a specific way

I think we can get rid of this and simply cast interface to specific type